fs-sim/fs-sim.cabal

```diff
@@ -30,7 +30,6 @@ library
     System.FS.Sim.Error
     System.FS.Sim.FsTree
     System.FS.Sim.MockFS
-    System.FS.Sim.Pure
```

Comment:
My thinking here is that we can safely remove this module and refrain from exposing `System.FS.Sim.Prim` because these modules are not currently used by `ouroboros-consensus` or `lsm-tree`. It's more useful for those packages to use `simHasFS` or `simErrorHasFS` instead. However, if in the future there would be a request to expose parts of the `System.FS.Sim.Prim` module, we can do that then. For now, I'm in favour of applying the YAGNI principle